### ISSUE TITLE
请教： torch.sum(moving_mean*nn.Softmax(-1)(self.layer(x.unsqueeze(-1))),dim=-1)

### ISSUE THREAD

**Issue by rouniuyizu** [MAZiqing/FEDformer#20]
请教下这里使用Softmax的原因：为什么不简单的对于不同的x-moving_mean求平均。

同时为什么使用x而不是moving_mean自身作为gate？

感谢

**Comment by tianzhou2011** [comment]
也没有特别的理由==你说的这些也都可以试一试，看看效果，我们用的是一种相对标准的MOE的设定，也是当时想到的一个，其实MOE算法相关的paper都有几百篇了吧，各种选择其实非常多，我们也没在这特别研究哪种最好。

**Comment by rouniuyizu** [comment]
了解了，谢谢
